movieApp.js

```diff
@@ -107,18 +101,149 @@ for (let movie of movies) {
   }
   createCommentSection();
 
-  readMoreBtn.addEventListener("click", function onclick() {
-    // location.href = "_blank";
-    // Create element to show details about the movie
-
-    /* Toggle between adding and removing the "active" class,
-     to highlight the button that controls the panel */
-    this.classList.toggle("active");
-    /* Toggle between hiding and showing the active panel */
-    if (movieDetails.style.display === "block") {
-      movieDetails.style.display = "none";
-    } else {
-      movieDetails.style.display = "block";
+  function toggle() {
+    {
+      this.classList.toggle("active");
+      /* Toggle between hiding and showing the active panel */
+      if (movieDetails.style.display === "block") {
+        movieDetails.style.display = "none";
+      } else {
+        movieDetails.style.display = "block";
+      }
     }
+  }
+
+  readMoreBtn.addEventListener("click", toggle);
+  return movieContainer;
+}
+
+function createMoviesPage() {
+  nowStreaming.innerHTML = "";
+  movies.forEach((movie) => {
+    const movieElement = createMovieCard(movie);
+    nowStreaming.appendChild(movieElement);
   });
 }
+createMoviesPage();
+// JS 2
+const searchBtn = document.getElementById("search-icon");
+
+// function to search
+const inputField = document.getElementById("keyword-search");
+searchBtn.addEventListener("click", () => {
+  const inputStr = inputField.value.toLowerCase();
+  nowStreaming.innerHTML = "";
+  let found = false; // Flag to check if any movie is found
+
+  movies.forEach((movie) => {
+    if (
+      movie.title.toLowerCase().includes(inputStr) ||
+      movie.description.toLowerCase().includes(inputStr)
+    ) {
+      found = true;
+      const searchMovieElement = createMovieCard(movie);
+      nowStreaming.appendChild(searchMovieElement);
+    }
+    if (!found) {
```

Comment:
Are you sure this should not be outside the for each?
I.e., you loop through and try to find the movie. If it is not found _after the loop is over_, you say "not found".

Reply:
yes, that was a mistake and now has been corrected.